adapters/pgamssp/pgamssp.go

```diff
@@ -0,0 +1,154 @@
+package pgamssp
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	PGAMSspBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var err error
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var pgamExt openrtb_ext.ImpExtPgamSsp
+
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
+			return nil, []error{err}
+		}
+		if err = json.Unmarshal(bidderExt.Bidder, &pgamExt); err != nil {
+			return nil, []error{err}
+		}
+
+		temp := reqBodyExt{PGAMSspBidderExt: reqBodyExtBidder{}}
+
+		if pgamExt.PlacementID != "" {
+			temp.PGAMSspBidderExt.PlacementID = pgamExt.PlacementID
+			temp.PGAMSspBidderExt.Type = "publisher"
+		} else if pgamExt.EndpointID != "" {
+			temp.PGAMSspBidderExt.EndpointID = pgamExt.EndpointID
+			temp.PGAMSspBidderExt.Type = "network"
+		}
```

Comment:
Confirming that you want to default type to publisher if both `PlacementID` and `EndpointID` is defined. Also requesting to please add a json test case for this behaviour.

Reply:
`{"placementId": "1", "endpointId": "1"}`,
these are invalid parameters, they cannot be both at the same time